File: Codes/Homework/euler's_method.py
import re

def clean(f):
    f_new = ""
    
    for i in range(len(f)):
        if f[i] == '^':
            f_new += "**"
        elif i > 0 and (f[i] == 'x' or f[i] == 'y'):
            if len(f_new) > 0 and f_new[-1] not in '+-*/ ':
                f_new += "*"
            f_new += f[i]
        else:
            f_new += f[i]
    
    return f_new

f = input("Enter expression:\n").split('=')[1]
h = float(input("Enter h:\n").split('=')[1])
val = float(input("Enter stop value:\n").split('=')[1])

initial = input("Enter initial value of y [y(x0) = val]:\n")
match = re.match(r'y\(([\d\.]+)\)\s*=\s*([\d\.]+)', initial)

x = float(match.group(1))
y = float(match.group(2))
f = clean(f)

print(f"f = {f}")

print(eval(f))

while x < val:
    print("x =", x, "y =", y)
    y = y + h * eval(f)
    x += h
    
print("x =", x, "y =", y)

File: Codes/Homework/euler's_method.c
#include <stdio.h>

double f(double x, double y)
{
    return 3 * x * x + 1;
}

int main()
{
    double x, y, h, stop;

    x = 1;
    y = 2;
    h = 0.25;
    stop = 2;

    scanf("%lf %lf %lf %lf", &x, &y, &h, &stop);

    printf("x = %g, y = %g\n", x, y);

    while (x < stop) {
        y = y + h * f(x, y);

        x += h;

        printf("x = %g, y = %g\n", x, y);
    }

    return 0;
}